src/main/catalog-sources/weblinks.ts

```diff
@@ -28,9 +28,12 @@ async function validateLink(link: WebLink) {
   }
 }
 
+interface Dependencies {
+  weblinkStore: WeblinkStore;
+  catalogEntityRegistry: CatalogEntityRegistry;
+}
 
-export function syncWeblinks() {
-  const weblinkStore = WeblinkStore.getInstance();
```

Comment:
Why not just make this fully injectable?

Reply:
`syncWebLinks` part done. WeblinkStore cascades a bit too much, so not including it to this PR.